stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/querychannels/logic/QueryChannelsLogic.kt

```diff
@@ -119,8 +119,10 @@ internal class QueryChannelsLogic(
     ) {
         if (isFirstPage) {
             (mutableState._channels.value - channels.map { it.cid }).values
-                .filterNot { mutableState.newChannelEventFilter(it, mutableState.filter) }
-                .map { it.cid }
+                .map(Channel::cid)
+                .filterNot { cid ->
+                    mutableState.defaultChannelEventsHandler.newChannelEventFilter(cid, mutableState.filter)
```

Comment:
This is very confusing - the function names don't really give much info as to what's happening.

The `filterNot` also makes it hard, as we're filtering the cases where a `cid` is not... what?

Not a `newChannelEvent`? I would rather have something like:

```
filterNot { cid ->
  val eventsHandler = ...

  eventsHandler.isNewChannelEvent(cid, filter)
}
```

Otherwise, we're just writing code that not very many people will understand and is hard to update.

Reply:
Actually I don't know why this piece of code exists. I touched it several times for refactoring purpose actually, like this one. But I'm not sure what happens if we delete this piece completely (suspect that nothing, but prefer to keep it as is currently)